dbms/src/Storages/DeltaMerge/tests/gtest_dm_delta_merge_store_for_fast_mode.cpp

```diff
@@ -0,0 +1,1620 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Exception.h>
+#include <Common/FailPoint.h>
+#include <Common/Logger.h>
+#include <DataStreams/BlocksListBlockInputStream.h>
+#include <DataStreams/OneBlockInputStream.h>
+#include <DataTypes/DataTypeString.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTLiteral.h>
+#include <Poco/File.h>
+#include <Storages/DeltaMerge/DMContext.h>
+#include <Storages/DeltaMerge/DeltaMergeStore.h>
+#include <Storages/DeltaMerge/File/DMFileBlockOutputStream.h>
+#include <Storages/DeltaMerge/Filter/RSOperator.h>
+#include <Storages/DeltaMerge/PKSquashingBlockInputStream.h>
+#include <Storages/DeltaMerge/Segment.h>
+#include <Storages/DeltaMerge/tests/DMTestEnv.h>
+#include <Storages/DeltaMerge/tests/MultiSegmentTestUtil.h>
+#include <Storages/tests/TiFlashStorageTestBasic.h>
+#include <TestUtils/FunctionTestUtils.h>
+#include <TestUtils/TiFlashTestBasic.h>
+#include <fmt/format.h>
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+/// This test file is mainly test on the correctness of read in fast mode.
+/// Because the basic functions are tested in gtest_dm_delta_merge_storage.cpp, we will not cover it here.
+
+namespace DB
+{
+namespace FailPoints
+{
+} // namespace FailPoints
+
+namespace DM
+{
+extern DMFilePtr writeIntoNewDMFile(DMContext & dm_context, //
+                                    const ColumnDefinesPtr & schema_snap,
+                                    const BlockInputStreamPtr & input_stream,
+                                    UInt64 file_id,
+                                    const String & parent_path,
+                                    DMFileBlockOutputStream::Flags flags);
+namespace tests
+{
+class DeltaMergeStoreTest : public DB::base::TiFlashStorageTestBasic
```

Comment:
I think we need a header file for class `DeltaMergeStoreTest`/`DeltaMergeStoreRWTest` since it is shared between this file and gtest_dm_delta_merge_storage.cpp